awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -117,22 +115,23 @@ def get_all_endpoint(self, endpoint, *args, **kwargs):
             response['json']['next'] = next_page
         return response
 
-    def get_one(self, endpoint, name_or_id=None, *args, **kwargs):
+    def get_one(self, endpoint, name_or_id=None, allow_none=True, **kwargs):
+        new_kwargs = kwargs.copy()
         if name_or_id:
             name_field = self.get_name_field_from_endpoint(endpoint)
-            new_args = kwargs.get('data', {}).copy()
-            if name_field in new_args:
+            new_data = kwargs.get('data', {}).copy()
+            if name_field in new_data:
                 self.fail_json(msg="You can't specify the field {0} in your search data if using the name_or_id field".format(name_field))
 
-            new_args['or__{0}'.format(name_field)] = name_or_id
             try:
-                new_args['or__id'] = int(name_or_id)
+                new_data['or__id'] = int(name_or_id)
+                new_data['or__{0}'.format(name_field)] = name_or_id
```

Comment:
@beeankha I made this change in part because it made the error message better.

But I also don't know how correct it is to do `/api/v2/inventories/?or__name=foo&organization=2`. If that's the query you're doing, it's better to just get rid of the `or__`, because that only matters if the id-based lookup is in play, and it's not here.